File: backend/src/main/java/com/shopcart/entity/Inventory.java
package com.shopcart.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "inventories")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Inventory {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "product_id", unique = true, nullable = false)
    private String productId;

    @Column(nullable = false)
    private Integer quantity; // total_stock

    @Column(name = "reserved_quantity", nullable = false)
    @Builder.Default
    private Integer reservedQuantity = 0;

    @Column(name = "sold_quantity", nullable = false)
    @Builder.Default
    private Integer soldQuantity = 0;
}

File: frontend/src/types/inventory.ts
// src/types/inventory.ts

export interface InventoryItem {
  id: string;
  productId: string;
  quantity: number; // Thực tế
  reservedQuantity: number; // Đang giữ
  soldQuantity: number; // Đã bán
  availableQuantity: number; // Đặt khả dụng
}
